l10n_nl_tax_declaration_reporting/report/report_vat.py

```diff
@@ -0,0 +1,213 @@
+        _3AO = self._get_line_value('3a', 'omzet', tax_report_lines)
+        _3BO = self._get_line_value('3b', 'omzet', tax_report_lines)
+        _3CO = self._get_line_value('3c', 'omzet', tax_report_lines)
+        _4AO = self._get_line_value('4a', 'omzet', tax_report_lines)
+        _4AB = self._get_line_value('4a', 'btw', tax_report_lines)
+        _4BO = self._get_line_value('4b', 'omzet', tax_report_lines)
+        _4BB = self._get_line_value('4b', 'btw', tax_report_lines)
+        _5BB = self._get_line_value('5b', 'btw', tax_report_lines)
+
+        _5AB = _1AB + _1BB + _1CB + _1DB + _2AB - _4AB - _4BB
+        _5CB = _5AB + _5BB
+
+        docargs = {
+            'doc_ids': self._ids,
+            'doc_model': report.model,
+            'docs': self,
+            'time_now': fields.datetime.now(),
+            'chart': _chart_tax_name,
+            'fiscalyear': _fiscalyear_name,
+            'start_period': _period_from_name,
+            'end_period': _period_to_name,
+            'basedon': data['form']['based_on'],
+            '_1AO': _1AO,
+            '_1AB': _1AB,
+            '_1BO': _1BO,
+            '_1BB': _1BB,
+            '_1CO': _1CO,
+            '_1CB': _1CB,
+            '_1DO': _1DO,
+            '_1DB': _1DB,
+            '_1EO': _1EO,
+            '_2AO': _2AO,
+            '_2AB': _2AB,
+            '_3AO': _3AO,
+            '_3BO': _3BO,
+            '_3CO': _3CO,
+            '_4AO': _4AO,
+            '_4AB': _4AB,
+            '_4BO': _4BO,
+            '_4BB': _4BB,
+            '_5AB': _5AB,
+            '_5BB': _5BB,
+            '_5CB': _5CB,
+            'tax_code_error': self.tax_code_error,
+        }
+
+        return report_obj.render(
+            'l10n_nl_tax_declaration_reporting.tax_report_template', docargs)
+
+    def _get_line_value(self, code, code_type, tax_report_lines):
+        try:
+            return filter(lambda x: x[0] == code,
+                          filter(lambda x: x[1] == code_type,
+                                 tax_report_lines))[0][2]
+        except:
+            if not self.tax_code_error:
+                self.tax_code_error = \
+                    "Please add (BTW) or (omzet) to tax code: " + code
+            else:
+                self.tax_code_error += " and " + code
+            return 0
+
+    def _get_lines(self, based_on, company_id=False,
+                   parent=False, level=0, context=None):
+        period_list = self.period_ids
+        res = self._get_codes(based_on, company_id, parent, level,
+                              period_list, context=context)
+        if period_list:
+            res = self._add_codes(based_on, res, period_list, context=context)
+        else:
+            self.cr.execute("select id from account_fiscalyear")
+            fy = self.cr.fetchall()
+            self.cr.execute(
+                "select id from account_period where fiscalyear_id = %s",
+                (fy[0][0],))
```

Comment:
Do you mean to select all periods from the current fiscal year? In that case, you need do make sure you select the right fiscal year (in this case the order seems to be undefined). Also, make the query company aware (or use the ORM to select the fiscal year, which applies standard company filters).